Add Mantine theme - rjsf-v6

zfarhad opened the pull request:
### Reasons for making this change

[Please describe them here]

Added a new theme using [Mantine UI](https://mantine.dev/) library (@mantine/core, @mantine/hooks, @mantine/dates) for react-jsonschema-form v6, following the PR https://github.com/rjsf-team/react-jsonschema-form/pull/4496 and addressing rjsf-v6.
This theme provides a modern, customizable UI component set that integrates well with RJSF's form generation capabilities.

The Mantine theme offers:

    Modern, clean UI components
    Built-in dark mode support
    Responsive design
    Accessible components
    Customizable styling through Mantine's theming system

### Checklist

- [ ] **I'm updating documentation**
  - [ ] I've [checked the rendering](https://rjsf-team.github.io/react-jsonschema-form/docs/contributing) of the Markdown text I've added
- [x] **I'm adding or updating code**
  - [x] I've added and/or updated tests. I've run `npx nx run-many --target=build --exclude=@rjsf/docs && npm run test:update` to update snapshots, if needed.
  - [x] I've updated [docs](https://rjsf-team.github.io/react-jsonschema-form/docs) if needed
  - [ ] I've updated the [changelog](https://github.com/rjsf-team/react-jsonschema-form/blob/main/CHANGELOG.md) with a description of the PR
- [ ] **I'm adding a new feature**
  - [ ] I've updated the playground with an example use of the feature

heath-freenome commented on packages/mantine/package.json:
```diff
@@ -0,0 +1,97 @@
+  "version": "6.0.0-alpha.0",
+  "main": "dist/index.js",
+  "module": "lib/index.js",
+  "typings": "lib/index.d.ts",
```
You will need to add this to match the other theme's improved ESM support:
```suggestion
  "typings": "lib/index.d.ts",
  "type": "module",
  "exports": {
    ".": {
      "require": "./dist/index.js",
      "import": "./lib/index.js",
      "types": "./lib/index.d.ts"
    },
    "./lib": {
      "require": "./dist/index.js",
      "import": "./lib/index.js",
      "types": "./lib/index.d.ts"
    },
    "./lib/*.js": {
      "require": "./dist/*.js",
      "import": "./lib/*.js",
      "types": "./lib/*.d.ts"
    },
    "./dist": {
      "require": "./dist/index.js",
      "import": "./lib/index.js",
      "types": "./lib/index.d.ts"
    },
    "./dist/*.js": {
      "require": "./dist/*.js",
      "import": "./lib/*.js",
      "types": "./lib/*.d.ts"
    }
  },
```

heath-freenome commented on packages/mantine/package.json:
```diff
@@ -0,0 +1,97 @@
+  ],
+  "engineStrict": false,
+  "engines": {
+    "node": ">=14"
```
we dropped node 14 support
```suggestion
    "node": ">=20"
```

heath-freenome commented on packages/mantine/src/templates/ArrayFieldItemTemplate.tsx:
```diff
@@ -0,0 +1,83 @@
+  );
+}
+
+/*
```
Please delete the unused comment

heath-freenome commented on packages/mantine/src/templates/ArrayFieldTemplate.tsx:
```diff
@@ -0,0 +1,101 @@
+      {canAdd && (
+        <Group justify='flex-end'>
+          <AddButton
+            className='array-item-add'
```
We added ids on all buttons:
```suggestion
            id={buttonId<T>(idSchema, 'add')}
            className='array-item-add'
```

heath-freenome commented on packages/mantine/src/templates/BaseInputTemplate.tsx:
```diff
@@ -0,0 +1,113 @@
+import { TextInput, NumberInput } from '@mantine/core';
+import { cleanupOptions } from '../utils';
+
+export default function BaseInputTemplate<
```
Can you add JSDocs please. Feel free to copy from another theme

heath-freenome commented on packages/mantine/src/templates/FieldTemplate.tsx:
```diff
@@ -0,0 +1,66 @@
+        </Text>
+      )}
+      {errors}
+      {help}
```
Did you want to add the remove button that other themes add here?

heath-freenome commented on packages/mantine/src/templates/ObjectFieldTemplate.tsx:
```diff
@@ -0,0 +1,95 @@
+      {canExpand(schema, uiSchema, formData) && (
+        <Box mt='xs'>
+          <AddButton
+            disabled={disabled || readonly}
```
Add the id and marker class
```suggestion
            id={buttonId<T>(idSchema, 'add')}
            className='object-property-expand'
            disabled={disabled || readonly}
```

heath-freenome commented on packages/mantine/src/widgets/CheckboxWidget.tsx:
```diff
@@ -0,0 +1,104 @@
+import { Checkbox } from '@mantine/core';
+import { cleanupOptions } from '../utils';
+
+export default function CheckboxWidget<
```
Add JSDoc please, feel free to copy from another theme

heath-freenome commented on packages/mantine/src/widgets/DateTime/AltDateTimeWidget.tsx:
```diff
@@ -0,0 +1,17 @@
+
+import _AltDateWidget from './AltDateWidget';
+
+export default function AltDateTimeWidget<
```
Add JSDoc please, feel free to copy from another theme

heath-freenome commented on packages/mantine/src/widgets/DateTime/AltDateWidget.tsx:
```diff
@@ -0,0 +1,138 @@
+/** The `AltDateWidget` is an alternative widget for rendering date properties.
+ * @param props - The `WidgetProps` for this component
+ */
+function AltDateWidget<T = any, S extends StrictRJSFSchema = RJSFSchema, F extends FormContextType = any>(
+  props: WidgetProps<T, S, F>
+) {
```
This is preferred
```suggestion
export default function AltDateWidget<T = any, S extends StrictRJSFSchema = RJSFSchema, F extends FormContextType = any>(
  props: WidgetProps<T, S, F>
) {
```

heath-freenome commented on packages/mantine/src/widgets/DateTime/AltDateWidget.tsx:
```diff
@@ -0,0 +1,138 @@
+
+AltDateWidget.defaultProps = {
+  showTime: false,
+};
+
+export default AltDateWidget;
```
```suggestion
};
```

heath-freenome commented on packages/mantine/src/widgets/DateTime/DateTimeInput.tsx:
```diff
@@ -0,0 +1,88 @@
+  return dayjs(date).format(format || 'YYYY-MM-DD');
+};
+
+export default function DateTimeInput<
```
Add JSDoc please, feel free to copy from another theme

heath-freenome commented on packages/mantine/src/templates/GridTemplate.tsx:
Can you update the `layoutGrid.tsx` in the playground to add an example for `mantine`. Thanks!

heath-freenome commented on packages/mantine/jest.config.js:
```diff
@@ -0,0 +1,10 @@
+module.exports = {
```
You will need to convert this to a `.json`  file due to the `package.json` changes I'm requesting

heath-freenome commented on packages/mantine/babel.config.js:
```diff
@@ -0,0 +1,3 @@
+const defaultConfig = require('../../babel.config');
```
You will need to convert this to a `.json`  file due to the `package.json` changes I'm requesting

heath-freenome commented on packages/mantine/package.json:
```diff
@@ -0,0 +1,97 @@
+    "node": ">=14"
+  },
+  "scripts": {
+    "build:ts": "tsc -b",
```
You will want to copy the `tsconfig.build.json` file from one of the other themes, like `chakra-ui`
```suggestion
    "build:ts": "tsc -b && tsc-alias -p tsconfig.build.json",
```

zfarhad commented:
@heath-freenome All requested changes have been applied, please check and let me know if there is anything else

heath-freenome commented on packages/mantine/src/templates/ButtonTemplates/IconButton.tsx:
```diff
@@ -0,0 +1,86 @@
+import React from 'react';
```
Rather than importing `React` can you instead import the elements of react instead?
```suggestion
import { MouseEventHandler } from 'react';
```

heath-freenome commented on packages/mantine/src/templates/BaseInputTemplate.tsx:
```diff
@@ -0,0 +1,119 @@
+import React from 'react';
```
Rather than importing from `React`, only import what you need
```suggestion
import { FocusEvent } from 'react';
```

heath-freenome commented on packages/mantine/src/templates/WrapIfAdditionalTemplate.tsx:
```diff
@@ -0,0 +1,96 @@
+import React from 'react';
```
Rather than importing from `React`, only import what you need
```suggestion
import { FocusEvent } from 'react';
```

heath-freenome commented on packages/mantine/src/templates/icons.tsx:
```diff
@@ -0,0 +1,141 @@
+import React from 'react';
```
Rather than importing from `React`, only import what you need
```suggestion
import { ComponentPropsWithoutRef } from 'react';
```

heath-freenome commented on packages/mantine/src/widgets/CheckboxWidget.tsx:
```diff
@@ -0,0 +1,109 @@
+import React, { useCallback } from 'react';
```
```suggestion
import { ChangeEvent, FocusEvent, ReactElement, useCallback } from 'react';
```

heath-freenome commented on packages/mantine/src/widgets/CheckboxesWidget.tsx:
```diff
@@ -0,0 +1,111 @@
+import React, { useCallback } from 'react';
```
```suggestion
import { FocusEvent, useCallback } from 'react';
```

heath-freenome commented on packages/mantine/src/widgets/ColorWidget.tsx:
```diff
@@ -0,0 +1,85 @@
+import React, { useCallback } from 'react';
```
```suggestion
import { FocusEvent, useCallback } from 'react';
```

heath-freenome commented on packages/mantine/src/widgets/DateTime/TimeWidget.tsx:
```diff
@@ -0,0 +1,83 @@
+import React, { useCallback } from 'react';
```
```suggestion
import { ChangeEvent, FocusEvent, useCallback } from 'react';
```

heath-freenome commented on packages/mantine/src/widgets/PasswordWidget.tsx:
```diff
@@ -0,0 +1,87 @@
+import React, { useCallback } from 'react';
```
```suggestion
import { ChangeEvent, FocusEvent, useCallback } from 'react';
```

heath-freenome commented on packages/mantine/src/widgets/PasswordWidget.tsx:
```diff
@@ -0,0 +1,87 @@
+  StrictRJSFSchema,
+  WidgetProps,
+} from '@rjsf/utils';
+import { PasswordInput } from '@mantine/core';
+import { cleanupOptions } from '../utils';
```
Please add blank line between system and local imports
```suggestion
import { PasswordInput } from '@mantine/core';

import { cleanupOptions } from '../utils';
```

heath-freenome commented on packages/mantine/src/widgets/SelectWidget.tsx:
```diff
@@ -0,0 +1,108 @@
+import React, { useCallback, useMemo } from 'react';
```
```suggestion
import { FocusEvent, useCallback, useMemo } from 'react';
```

heath-freenome commented on packages/mantine/src/widgets/SelectWidget.tsx:
```diff
@@ -0,0 +1,108 @@
+  StrictRJSFSchema,
+  WidgetProps,
+} from '@rjsf/utils';
+import { Select, MultiSelect } from '@mantine/core';
+import { cleanupOptions } from '../utils';
```
```suggestion
import { Select, MultiSelect } from '@mantine/core';

import { cleanupOptions } from '../utils';
```

heath-freenome commented on packages/mantine/src/widgets/TextareaWidget.tsx:
```diff
@@ -0,0 +1,78 @@
+import React, { useCallback } from 'react';
+import { StrictRJSFSchema, RJSFSchema, FormContextType, WidgetProps, labelValue } from '@rjsf/utils';
+import { Textarea } from '@mantine/core';
+import { cleanupOptions } from '../utils';
```
```suggestion
import { Textarea } from '@mantine/core';

import { cleanupOptions } from '../utils';
```

heath-freenome commented on packages/mantine/src/widgets/TextareaWidget.tsx:
```diff
@@ -0,0 +1,78 @@
+import React, { useCallback } from 'react';
```
```suggestion
import { ChangeEvent, FocusEvent, useCallback } from 'react';
```

heath-freenome commented on packages/mantine/src/templates/BaseInputTemplate.tsx:
```diff
@@ -0,0 +1,119 @@
+  RJSFSchema,
+  StrictRJSFSchema,
+} from '@rjsf/utils';
+import { TextInput, NumberInput } from '@mantine/core';
+import { cleanupOptions } from '../utils';
```
```suggestion
import { TextInput, NumberInput } from '@mantine/core';

import { cleanupOptions } from '../utils';
```

heath-freenome commented on packages/mantine/src/widgets/CheckboxWidget.tsx:
```diff
@@ -0,0 +1,109 @@
+  labelValue,
+  ariaDescribedByIds,
+} from '@rjsf/utils';
+import { Checkbox } from '@mantine/core';
+import { cleanupOptions } from '../utils';
```
```suggestion
import { Checkbox } from '@mantine/core';

import { cleanupOptions } from '../utils';
```

heath-freenome commented on packages/mantine/src/widgets/CheckboxesWidget.tsx:
```diff
@@ -0,0 +1,111 @@
+  RJSFSchema,
+  StrictRJSFSchema,
+} from '@rjsf/utils';
+import { Checkbox, Flex, Input } from '@mantine/core';
+import { cleanupOptions } from '../utils';
```
```suggestion
import { Checkbox, Flex, Input } from '@mantine/core';

import { cleanupOptions } from '../utils';
```

heath-freenome commented on packages/mantine/src/widgets/ColorWidget.tsx:
```diff
@@ -0,0 +1,85 @@
+  labelValue,
+  ariaDescribedByIds,
+} from '@rjsf/utils';
+import { ColorInput } from '@mantine/core';
+import { cleanupOptions } from '../utils';
```
```suggestion
import { ColorInput } from '@mantine/core';

import { cleanupOptions } from '../utils';
```

heath-freenome commented on packages/mantine/src/widgets/FileWidget.tsx:
```diff
@@ -0,0 +1,174 @@
+  StrictRJSFSchema,
+  WidgetProps,
+} from '@rjsf/utils';
+import { FileInput, Pill } from '@mantine/core';
+import { cleanupOptions } from '../utils';
```
```suggestion
import { FileInput, Pill } from '@mantine/core';

import { cleanupOptions } from '../utils';
```

heath-freenome commented on packages/mantine/src/widgets/RadioWidget.tsx:
```diff
@@ -0,0 +1,102 @@
+  StrictRJSFSchema,
+  WidgetProps,
+} from '@rjsf/utils';
+import { Radio, Flex } from '@mantine/core';
+import { cleanupOptions } from '../utils';
```
```suggestion
import { Radio, Flex } from '@mantine/core';

import { cleanupOptions } from '../utils';
```

heath-freenome commented on packages/mantine/src/widgets/RangeWidget.tsx:
```diff
@@ -0,0 +1,86 @@
+  WidgetProps,
+  titleId,
+} from '@rjsf/utils';
+import { Slider, Input } from '@mantine/core';
+import { cleanupOptions } from '../utils';
```
```suggestion
import { Slider, Input } from '@mantine/core';

import { cleanupOptions } from '../utils';
```

heath-freenome commented on packages/mantine/src/widgets/index.ts:
```diff
@@ -0,0 +1,42 @@
+import { FormContextType, RegistryWidgetsType, RJSFSchema, StrictRJSFSchema } from '@rjsf/utils';
+
+import dayjs from 'dayjs';
```
```suggestion
import { FormContextType, RegistryWidgetsType, RJSFSchema, StrictRJSFSchema } from '@rjsf/utils';
import dayjs from 'dayjs';
```

heath-freenome commented on packages/mantine/src/widgets/index.ts:
```diff
@@ -0,0 +1,42 @@
+
+import dayjs from 'dayjs';
+import customParseFormat from 'dayjs/plugin/customParseFormat';
+// This plugin is needed to support the parsing of date and time values in the `DateWidget` and `DateTimeWidget`
+dayjs.extend(customParseFormat);
```
Can you move this code after the imports

heath-freenome commented:
@zfarhad Can you also fix the build issues

zfarhad commented:
@heath-freenome All the code cleanups applied

I tried to fix the build, but I get this error for all packages, not just mantine:

`
      > @rjsf/mantine@6.0.0-alpha.0 build
      > npm run build:ts && npm run build:cjs && npm run build:esm && npm run build:umd
      
      
      > @rjsf/mantine@6.0.0-alpha.0 build:ts
      > tsc -b tsconfig.build.json && tsc-alias -p tsconfig.build.json
      
      ../utils/src/ErrorSchemaBuilder.ts(43,45): error TS2590: Expression produces a union type that is too complex to represent.
      ../utils/src/getOptionMatchingSimpleDiscriminator.ts(31,25): error TS2339: Property 'type' does not exist on type 'JSONSchema7Definition | {}'.
        Property 'type' does not exist on type 'false'.
      ../utils/src/getOptionMatchingSimpleDiscriminator.ts(31,60): error TS2339: Property 'type' does not exist on type 'JSONSchema7Definition | {}'.
        Property 'type' does not exist on type 'false'.
      ../utils/src/getOptionMatchingSimpleDiscriminator.ts(35,25): error TS2339: Property 'const' does not exist on type 'JSONSchema7Definition | {}'.
        Property 'const' does not exist on type 'false'.
      ../utils/src/getOptionMatchingSimpleDiscriminator.ts(39,25): error TS2339: Property 'enum' does not exist on type 'JSONSchema7Definition | {}'.
        Property 'enum' does not exist on type 'false'.
      ../utils/src/schema/getMatchingOption.ts(50,29): error TS2345: Argument of type '{} | Exclude<NonNullable<S["properties"]>[string], undefined>' is not assignable to parameter of type 'S'.
        'S' could be instantiated with an arbitrary type which could be unrelated to '{} | Exclude<NonNullable<S["properties"]>[string], undefined>'.
      ../utils/src/optionsList.ts(73,15): error TS2322: Type 'NonNullable<S["properties"]>[string] | undefined' is not assignable to type 'S'.
        'S' could be instantiated with an arbitrary type which could be unrelated to 'NonNullable<S["properties"]>[string] | undefined'.
      ../utils/src/schema/getDefaultFormState.ts(444,54): error TS2322: Type 'NonNullable<S["properties"]>[string]' is not assignable to type 'object'.
        Type 'JSONSchema7Definition' is not assignable to type 'object'.
          Type 'boolean' is not assignable to type 'object'.
      ../utils/src/schema/getDefaultFormState.ts(444,54): error TS18048: 'propertySchema' is possibly 'undefined'.
      ../utils/src/schema/getDefaultFormState.ts(446,36): error TS2345: Argument of type 'JSONSchema7Definition | undefined' is not assignable to parameter of type 'JSONSchema7'.
        Type 'undefined' is not assignable to type 'JSONSchema7'.
      ../utils/src/schema/getDefaultFormState.ts(449,69): error TS2345: Argument of type 'NonNullable<S["properties"]>[string] | undefined' is not assignable to parameter of type 'S'.
        'S' could be instantiated with an arbitrary type which could be unrelated to 'NonNullable<S["properties"]>[string] | undefined'.
      ../utils/src/schema/sanitizeDataForNewSchema.ts(96,11): error TS2322: Type '{} | Exclude<NonNullable<S["properties"]>[string], undefined>' is not assignable to type 'S'.
        'S' could be instantiated with an arbitrary type which could be unrelated to '{} | Exclude<NonNullable<S["properties"]>[string], undefined>'.
      ../utils/src/schema/sanitizeDataForNewSchema.ts(97,11): error TS2322: Type '{} | Exclude<NonNullable<(S & { properties: Record<string, unknown> extends S | undefined ? (S | undefined)[never] : unknown; } & { [uniqueSymbol]: unknown; })["properties"]>[string], undefined>' is not assignable to type 'S'.
        'S' could be instantiated with an arbitrary type which could be unrelated to '{} | Exclude<NonNullable<(S & { properties: Record<string, unknown> extends S | undefined ? (S | undefined)[never] : unknown; } & { [uniqueSymbol]: unknown; })["properties"]>[string], undefined>'.
      ../utils/src/schema/toIdSchema.ts(81,9): error TS2345: Argument of type 'NonNullable<S["properties"]>[string] | {} | undefined' is not assignable to parameter of type 'S'.
        'S' could be instantiated with an arbitrary type which could be unrelated to 'NonNullable<S["properties"]>[string] | {} | undefined'.
      ../utils/src/schema/toPathSchema.ts(149,9): error TS2345: Argument of type 'NonNullable<S["properties"]>[string] | undefined' is not assignable to parameter of type 'S'.
        'S' could be instantiated with an arbitrary type which could be unrelated to 'NonNullable<S["properties"]>[string] | undefined'.
      ../validator-ajv8/src/createAjvInstance.ts(4,75): error TS2307: Cannot find module '@rjsf/utils' or its corresponding type declarations.
      ../validator-ajv8/src/createAjvInstance.ts(43,16): error TS2345: Argument of type 'import("/projects/react-jsonschema-form/packages/validator-ajv8/node_modules/ajv/dist/ajv").default' is not assignable to parameter of type 'import("/projects/react-jsonschema-form/node_modules/ajv-formats/node_modules/ajv/dist/core").default'.
        Types of property 'opts' are incompatible.
          Type 'import("/projects/react-jsonschema-form/packages/validator-ajv8/node_modules/ajv/dist/core").InstanceOptions' is not assignable to type 'import("/projects/react-jsonschema-form/node_modules/ajv-formats/node_modules/ajv/dist/core").InstanceOptions'.
            Type 'InstanceOptions' is not assignable to type 'CurrentOptions'.
              Types of property 'keywords' are incompatible.
                Type 'import("/projects/react-jsonschema-form/packages/validator-ajv8/node_modules/ajv/dist/types/index").Vocabulary | undefined' is not assignable to type 'import("/projects/react-jsonschema-form/node_modules/ajv-formats/node_modules/ajv/dist/types/index").Vocabulary | undefined'.
                  Type 'Vocabulary' is not assignable to type '(string | KeywordDefinition)[]'.
                    Type 'string | import("/projects/react-jsonschema-form/packages/validator-ajv8/node_modules/ajv/dist/types/index").KeywordDefinition' is not assignable to type 'string | import("/projects/react-jsonschema-form/node_modules/ajv-formats/node_modules/ajv/dist/types/index").KeywordDefinition'.
                      Type 'CodeKeywordDefinition' is not assignable to type 'string | KeywordDefinition'.
                        Type 'import("/projects/react-jsonschema-form/packages/validator-ajv8/node_modules/ajv/dist/types/index").CodeKeywordDefinition' is not assignable to type 'import("/projects/react-jsonschema-form/node_modules/ajv-formats/node_modules/ajv/dist/types/index").CodeKeywordDefinition'.
                          Types of property 'code' are incompatible.
                            Type '(cxt: import("/projects/react-jsonschema-form/packages/validator-ajv8/node_modules/ajv/dist/compile/validate/index").KeywordCxt, ruleType?: string | undefined) => void' is not assignable to type '(cxt: import("/projects/react-jsonschema-form/node_modules/ajv-formats/node_modules/ajv/dist/compile/validate/index").KeywordCxt, ruleType?: string | undefined) => void'.
                              Types of parameters 'cxt' and 'cxt' are incompatible.
                                Type 'import("/projects/react-jsonschema-form/node_modules/ajv-formats/node_modules/ajv/dist/compile/validate/index").KeywordCxt' is not assignable to type 'import("/projects/react-jsonschema-form/packages/validator-ajv8/node_modules/ajv/dist/compile/validate/index").KeywordCxt'.
                                  The types of 'gen._scope' are incompatible between these types.
                                    Type 'import("/projects/react-jsonschema-form/node_modules/ajv-formats/node_modules/ajv/dist/compile/codegen/scope").Scope' is not assignable to type 'import("/projects/react-jsonschema-form/packages/validator-ajv8/node_modules/ajv/dist/compile/codegen/scope").Scope'.
                                      Property '_names' is protected but type 'Scope' is not a class derived from 'Scope'.
      ../validator-ajv8/src/createAjvInstance.ts(45,16): error TS2345: Argument of type 'import("/projects/react-jsonschema-form/packages/validator-ajv8/node_modules/ajv/dist/ajv").default' is not assignable to parameter of type 'import("/projects/react-jsonschema-form/node_modules/ajv-formats/node_modules/ajv/dist/core").default'.
      ../validator-ajv8/src/compileSchemaValidatorsCode.ts(2,60): error TS2307: Cannot find module '@rjsf/utils' or its corresponding type declarations.
      ../validator-ajv8/src/compileSchemaValidatorsCode.ts(31,71): error TS2345: Argument of type '{ code: { source: boolean; es5?: boolean | undefined; esm?: boolean | undefined; lines: boolean; optimize?: number | boolean | undefined; formats?: Code | undefined; process?: ((code: string, schema?: SchemaEnv | undefined) => string) | undefined; regExp?: RegExpEngine | undefined; }; ... 46 more ...; unicode?: bool...' is not assignable to parameter of type 'Options'.
        Type '{ code: { source: boolean; es5?: boolean | undefined; esm?: boolean | undefined; lines: boolean; optimize?: number | boolean | undefined; formats?: Code | undefined; process?: ((code: string, schema?: SchemaEnv | undefined) => string) | undefined; regExp?: RegExpEngine | undefined; }; ... 46 more ...; unicode?: bool...' is not assignable to type 'CurrentOptions'.
          Types of property 'schemas' are incompatible.
            Type 'unknown[]' is not assignable to type 'AnySchema[] | { [x: string]: AnySchema | undefined; } | undefined'.
              Type 'unknown[]' is not assignable to type 'AnySchema[]'.
                Type 'unknown' is not assignable to type 'AnySchema'.
      ../validator-ajv8/src/compileSchemaValidators.ts(2,46): error TS2307: Cannot find module '@rjsf/utils' or its corresponding type declarations.
      ../validator-ajv8/src/processRawValidationErrors.ts(19,8): error TS2307: Cannot find module '@rjsf/utils' or its corresponding type declarations.
      ../validator-ajv8/src/precompiledValidator.ts(19,8): error TS2307: Cannot find module '@rjsf/utils' or its corresponding type declarations.
      ../validator-ajv8/src/createPrecompiledValidator.ts(1,78): error TS2307: Cannot find module '@rjsf/utils' or its corresponding type declarations.
      ../validator-ajv8/src/validator.ts(18,8): error TS2307: Cannot find module '@rjsf/utils' or its corresponding type declarations.
      ../validator-ajv8/src/customizeValidator.ts(1,63): error TS2307: Cannot find module '@rjsf/utils' or its corresponding type declarations.
      ../core/src/components/fields/ArrayField.tsx(20,8): error TS2307: Cannot find module '@rjsf/utils' or its corresponding type declarations.
      ../core/src/components/fields/BooleanField.tsx(11,8): error TS2307: Cannot find module '@rjsf/utils' or its corresponding type declarations.
      ../core/src/components/fields/BooleanField.tsx(58,17): error TS7006: Parameter 'option' implicitly has an 'any' type.
      ../core/src/components/fields/BooleanField.tsx(62,31): error TS2339: Property 'title' does not exist on type 'object'.
      ../core/src/components/fields/BooleanField.tsx(62,48): error TS2339: Property 'const' does not exist on type 'object'.
      ../core/src/components/fields/LayoutGridField.tsx(26,8): error TS2307: Cannot find module '@rjsf/utils' or its corresponding type declarations.
      ../core/src/components/fields/LayoutGridField.tsx(470,38): error TS2322: Type '{ className: any; }' is not assignable to type 'GridProps'.
        Object literal may only specify known properties, and 'className' does not exist in type 'GridProps'.
      ../core/src/components/fields/LayoutGridField.tsx(536,43): error TS18048: 'schema' is possibly 'undefined'.
      ../core/src/components/fields/LayoutGridField.tsx(542,9): error TS2322: Type 'S[[any, string]]' is not assignable to type 'S'.
        'S[[any, string]]' is assignable to the constraint of type 'S', but 'S' could be instantiated with a different subtype of constraint 'StrictRJSFSchema'.
      ../core/src/components/fields/LayoutGridField.tsx(564,39): error TS18048: 'schema' is possibly 'undefined'.
      ../core/src/components/fields/LayoutGridField.tsx(583,65): error TS2339: Property 'required' does not exist on type 'S & ({} | null)'.
        Property 'required' does not exist on type 'S & null'.
      ../core/src/components/fields/LayoutGridField.tsx(583,94): error TS2339: Property 'required' does not exist on type 'S & ({} | null)'.
        Property 'required' does not exist on type 'S & null'.
      ../core/src/components/fields/LayoutGridField.tsx(683,15): error TS2339: Property 'onChange' does not exist on type 'Readonly<LayoutGridFieldProps<T, S, F>>'.
      ../core/src/components/fields/LayoutGridField.tsx(683,25): error TS2339: Property 'errorSchema' does not exist on type 'Readonly<LayoutGridFieldProps<T, S, F>>'.
      ../core/src/components/fields/LayoutGridField.tsx(683,38): error TS2339: Property 'formData' does not exist on type 'Readonly<LayoutGridFieldProps<T, S, F>>'.
      ../core/src/components/fields/LayoutGridField.tsx(704,13): error TS2339: Property 'formData' does not exist on type 'Readonly<LayoutGridFieldProps<T, S, F>>'.
      ../core/src/components/fields/LayoutGridField.tsx(704,23): error TS2339: Property 'registry' does not exist on type 'Readonly<LayoutGridFieldProps<T, S, F>>'.
      ../core/src/components/fields/LayoutGridField.tsx(726,13): error TS2339: Property 'registry' does not exist on type 'Readonly<LayoutGridFieldProps<T, S, F>>'.
      ../core/src/components/fields/LayoutGridField.tsx(726,23): error TS2339: Property 'uiSchema' does not exist on type 'Readonly<LayoutGridFieldProps<T, S, F>>'.
      ../core/src/components/fields/LayoutGridField.tsx(750,13): error TS2339: Property 'registry' does not exist on type 'Readonly<LayoutGridFieldProps<T, S, F>>'.
      ../core/src/components/fields/LayoutGridField.tsx(750,23): error TS2339: Property 'uiSchema' does not exist on type 'Readonly<LayoutGridFieldProps<T, S, F>>'.
      ../core/src/components/fields/LayoutGridField.tsx(779,13): error TS2339: Property 'registry' does not exist on type 'Readonly<LayoutGridFieldProps<T, S, F>>'.
      ../core/src/components/fields/LayoutGridField.tsx(779,23): error TS2339: Property 'uiSchema' does not exist on type 'Readonly<LayoutGridFieldProps<T, S, F>>'.
      ../core/src/components/fields/LayoutGridField.tsx(805,13): error TS2339: Property 'registry' does not exist on type 'Readonly<LayoutGridFieldProps<T, S, F>>'.
      ../core/src/components/fields/LayoutGridField.tsx(805,23): error TS2339: Property 'schema' does not exist on type 'Readonly<LayoutGridFieldProps<T, S, F>>'.
      ../core/src/components/fields/LayoutGridField.tsx(805,42): error TS2339: Property 'formData' does not exist on type 'Readonly<LayoutGridFieldProps<T, S, F>>'.
      ../core/src/components/fields/LayoutGridField.tsx(813,9): error TS2322: Type '{ key: string; schema: any; layoutGridSchema: LayoutGridSchemaType; }' is not assignable to type 'IntrinsicAttributes & IntrinsicClassAttributes<LayoutGridField<T, S, F>> & Pick<Readonly<LayoutGridFieldProps<T, S, F>>, never> & InexactPartial<...> & InexactPartial<...>'.
        Property 'schema' does not exist on type 'IntrinsicAttributes & IntrinsicClassAttributes<LayoutGridField<T, S, F>> & Pick<Readonly<LayoutGridFieldProps<T, S, F>>, never> & InexactPartial<...> & InexactPartial<...>'.
      ../core/src/components/fields/LayoutGridField.tsx(836,7): error TS2339: Property 'schema' does not exist on type 'Readonly<LayoutGridFieldProps<T, S, F>>'.
      ../core/src/components/fields/LayoutGridField.tsx(837,7): error TS2339: Property 'uiSchema' does not exist on type 'Readonly<LayoutGridFieldProps<T, S, F>>'.
      ../core/src/components/fields/LayoutGridField.tsx(838,7): error TS2339: Property 'errorSchema' does not exist on type 'Readonly<LayoutGridFieldProps<T, S, F>>'.
      ../core/src/components/fields/LayoutGridField.tsx(839,7): error TS2339: Property 'idSchema' does not exist on type 'Readonly<LayoutGridFieldProps<T, S, F>>'.
      ../core/src/components/fields/LayoutGridField.tsx(840,7): error TS2339: Property 'onBlur' does not exist on type 'Readonly<LayoutGridFieldProps<T, S, F>>'.
      ../core/src/components/fields/LayoutGridField.tsx(841,7): error TS2339: Property 'onFocus' does not exist on type 'Readonly<LayoutGridFieldProps<T, S, F>>'.
      ../core/src/components/fields/LayoutGridField.tsx(842,7): error TS2339: Property 'formData' does not exist on type 'Readonly<LayoutGridFieldProps<T, S, F>>'.
      ../core/src/components/fields/LayoutGridField.tsx(843,7): error TS2339: Property 'readonly' does not exist on type 'Readonly<LayoutGridFieldProps<T, S, F>>'.
      ../core/src/components/fields/LayoutGridField.tsx(844,7): error TS2339: Property 'registry' does not exist on type 'Readonly<LayoutGridFieldProps<T, S, F>>'.
      ../core/src/components/fields/LayoutGridField.tsx(845,7): error TS2339: Property 'idSeparator' does not exist on type 'Readonly<LayoutGridFieldProps<T, S, F>>'.
      ../core/src/components/fields/LayoutGridField.tsx(944,13): error TS2339: Property 'uiSchema' does not exist on type 'Readonly<LayoutGridFieldProps<T, S, F>>'.
      ../core/src/components/fields/LayoutHeaderField.tsx(10,8): error TS2307: Cannot find module '@rjsf/utils' or its corresponding type declarations.
      ../core/src/components/fields/LayoutMultiSchemaField.tsx(23,8): error TS2307: Cannot find module '@rjsf/utils' or its corresponding type declarations.
      ../core/src/components/fields/MultiSchemaField.tsx(20,8): error TS2307: Cannot find module '@rjsf/utils' or its corresponding type declarations.
      ../core/src/components/fields/NumberField.tsx(2,98): error TS2307: Cannot find module '@rjsf/utils' or its corresponding type declarations.
      ../core/src/components/fields/ObjectField.tsx(19,8): error TS2307: Cannot find module '@rjsf/utils' or its corresponding type declarations.
      ../core/src/components/fields/SchemaField.tsx(22,8): error TS2307: Cannot find module '@rjsf/utils' or its corresponding type declarations.
      ../core/src/components/fields/SchemaField.tsx(316,40): error TS7006: Parameter '_schema' implicitly has an 'any' type.
      ../core/src/components/fields/SchemaField.tsx(340,40): error TS7006: Parameter '_schema' implicitly has an 'any' type.
      ../core/src/components/fields/StringField.tsx(10,8): error TS2307: Cannot find module '@rjsf/utils' or its corresponding type declarations.
      ../core/src/components/fields/NullField.tsx(2,75): error TS2307: Cannot find module '@rjsf/utils' or its corresponding type declarations.
      ../core/src/components/fields/index.ts(1,90): error TS2307: Cannot find module '@rjsf/utils' or its corresponding type declarations.
      ../core/src/components/templates/ArrayFieldDescriptionTemplate.tsx(9,8): error TS2307: Cannot find module '@rjsf/utils' or its corresponding type declarations.
      ../core/src/components/templates/ArrayFieldItemTemplate.tsx(9,8): error TS2307: Cannot find module '@rjsf/utils' or its corresponding type declarations.
      ../core/src/components/templates/ArrayFieldItemButtonsTemplate.tsx(8,8): error TS2307: Cannot find module '@rjsf/utils' or its corresponding type declarations.
      ../core/src/components/templates/ArrayFieldTemplate.tsx(10,8): error TS2307: Cannot find module '@rjsf/utils' or its corresponding type declarations.
      ../core/src/components/templates/ArrayFieldTitleTemplate.tsx(10,8): error TS2307: Cannot find module '@rjsf/utils' or its corresponding type declarations.
      ../core/src/components/templates/BaseInputTemplate.tsx(10,8): error TS2307: Cannot find module '@rjsf/utils' or its corresponding type declarations.
      ../core/src/components/templates/ButtonTemplates/SubmitButton.tsx(1,106): error TS2307: Cannot find module '@rjsf/utils' or its corresponding type declarations.
      ../core/src/components/templates/ButtonTemplates/IconButton.tsx(1,100): error TS2307: Cannot find module '@rjsf/utils' or its corresponding type declarations.
      ../core/src/components/templates/ButtonTemplates/AddButton.tsx(1,100): error TS2307: Cannot find module '@rjsf/utils' or its corresponding type declarations.
      ../core/src/components/templates/ButtonTemplates/index.ts(1,78): error TS2307: Cannot find module '@rjsf/utils' or its corresponding type declarations.
      ../core/src/components/templates/DescriptionField.tsx(1,86): error TS2307: Cannot find module '@rjsf/utils' or its corresponding type declarations.
      ../core/src/components/templates/ErrorList.tsx(8,8): error TS2307: Cannot find module '@rjsf/utils' or its corresponding type declarations.
      ../core/src/components/templates/FieldTemplate/FieldTemplate.tsx(8,8): error TS2307: Cannot find module '@rjsf/utils' or its corresponding type declarations.
      ../core/src/components/templates/FieldErrorTemplate.tsx(1,89): error TS2307: Cannot find module '@rjsf/utils' or its corresponding type declarations.
      ../core/src/components/templates/FieldErrorTemplate.tsx(22,20): error TS7006: Parameter 'elem' implicitly has an 'any' type.
      ../core/src/components/templates/FieldErrorTemplate.tsx(23,17): error TS7006: Parameter 'error' implicitly has an 'any' type.
      ../core/src/components/templates/FieldHelpTemplate.tsx(1,87): error TS2307: Cannot find module '@rjsf/utils' or its corresponding type declarations.
      ../core/src/components/templates/GridTemplate.tsx(1,35): error TS2307: Cannot find module '@rjsf/utils' or its corresponding type declarations.
      ../core/src/components/templates/ObjectFieldTemplate.tsx(13,8): error TS2307: Cannot find module '@rjsf/utils' or its corresponding type declarations.
      ../core/src/components/templates/TitleField.tsx(1,80): error TS2307: Cannot find module '@rjsf/utils' or its corresponding type declarations.
      ../core/src/components/templates/UnsupportedField.tsx(1,106): error TS2307: Cannot find module '@rjsf/utils' or its corresponding type declarations.
      ../core/src/components/templates/WrapIfAdditionalTemplate.tsx(9,8): error TS2307: Cannot find module '@rjsf/utils' or its corresponding type declarations.
      ../core/src/components/templates/index.ts(1,78): error TS2307: Cannot find module '@rjsf/utils' or its corresponding type declarations.
      ../core/src/components/widgets/AltDateWidget.tsx(15,8): error TS2307: Cannot find module '@rjsf/utils' or its corresponding type declarations.
      ../core/src/components/widgets/AltDateWidget.tsx(138,14): error TS7006: Parameter 'elemProps' implicitly has an 'any' type.
      ../core/src/components/widgets/AltDateWidget.tsx(138,25): error TS7006: Parameter 'i' implicitly has an 'any' type.
      ../core/src/components/widgets/AltDateTimeWidget.tsx(1,76): error TS2307: Cannot find module '@rjsf/utils' or its corresponding type declarations.
      ../core/src/components/widgets/CheckboxWidget.tsx(12,8): error TS2307: Cannot find module '@rjsf/utils' or its corresponding type declarations.
      ../core/src/components/widgets/CheckboxesWidget.tsx(13,8): error TS2307: Cannot find module '@rjsf/utils' or its corresponding type declarations.
      ../core/src/components/widgets/ColorWidget.tsx(1,89): error TS2307: Cannot find module '@rjsf/utils' or its corresponding type declarations.
      ../core/src/components/widgets/DateWidget.tsx(2,89): error TS2307: Cannot find module '@rjsf/utils' or its corresponding type declarations.
      ../core/src/components/widgets/DateTimeWidget.tsx(9,8): error TS2307: Cannot find module '@rjsf/utils' or its corresponding type declarations.
      ../core/src/components/widgets/DateTimeWidget.tsx(28,18): error TS7006: Parameter 'value' implicitly has an 'any' type.
      ../core/src/components/widgets/EmailWidget.tsx(1,89): error TS2307: Cannot find module '@rjsf/utils' or its corresponding type declarations.
      ../core/src/components/widgets/FileWidget.tsx(12,8): error TS2307: Cannot find module '@rjsf/utils' or its corresponding type declarations.
      ../core/src/components/widgets/HiddenWidget.tsx(1,76): error TS2307: Cannot find module '@rjsf/utils' or its corresponding type declarations.
      ../core/src/components/widgets/PasswordWidget.tsx(1,89): error TS2307: Cannot find module '@rjsf/utils' or its corresponding type declarations.
      ../core/src/components/widgets/RadioWidget.tsx(11,8): error TS2307: Cannot find module '@rjsf/utils' or its corresponding type declarations.
      ../core/src/components/widgets/RangeWidget.tsx(1,76): error TS2307: Cannot find module '@rjsf/utils' or its corresponding type declarations.
      ../core/src/components/widgets/SelectWidget.tsx(10,8): error TS2307: Cannot find module '@rjsf/utils' or its corresponding type declarations.
      ../core/src/components/widgets/TextareaWidget.tsx(2,96): error TS2307: Cannot find module '@rjsf/utils' or its corresponding type declarations.
      ../core/src/components/widgets/TextWidget.tsx(1,89): error TS2307: Cannot find module '@rjsf/utils' or its corresponding type declarations.
      ../core/src/components/widgets/TimeWidget.tsx(2,89): error TS2307: Cannot find module '@rjsf/utils' or its corresponding type declarations.
      ../core/src/components/widgets/URLWidget.tsx(1,89): error TS2307: Cannot find module '@rjsf/utils' or its corresponding type declarations.
      ../core/src/components/widgets/UpDownWidget.tsx(1,89): error TS2307: Cannot find module '@rjsf/utils' or its corresponding type declarations.
      ../core/src/components/widgets/index.ts(1,84): error TS2307: Cannot find module '@rjsf/utils' or its corresponding type declarations.
      ../core/src/getDefaultRegistry.ts(1,98): error TS2307: Cannot find module '@rjsf/utils' or its corresponding type declarations.
      ../core/src/components/Form.tsx(40,8): error TS2307: Cannot find module '@rjsf/utils' or its corresponding type declarations.
      ../core/src/withTheme.tsx(3,63): error TS2307: Cannot find module '@rjsf/utils' or its corresponding type declarations.
      ../core/src/withTheme.tsx(17,3): error TS2322: Type 'ForwardRefExoticComponent<Omit<FormProps<T, S, F>, "ref"> & RefAttributes<Form<T, S, F>>>' is not assignable to type 'ComponentType<FormProps<T, S, F>>'.
        Type 'ForwardRefExoticComponent<Omit<FormProps<T, S, F>, "ref"> & RefAttributes<Form<T, S, F>>>' is not assignable to type 'FunctionComponent<FormProps<T, S, F>>'.
          Types of property 'propTypes' are incompatible.
            Type 'WeakValidationMap<Omit<FormProps<T, S, F>, "ref"> & RefAttributes<Form<T, S, F>>> | undefined' is not assignable to type 'WeakValidationMap<FormProps<T, S, F>> | undefined'.
              Type 'WeakValidationMap<Omit<FormProps<T, S, F>, "ref"> & RefAttributes<Form<T, S, F>>>' is not assignable to type 'WeakValidationMap<FormProps<T, S, F>>'.
                Types of property 'ref' are incompatible.
                  Type 'Validator<LegacyRef<Form<T, S, F>> | undefined> | undefined' is not assignable to type 'Validator<Ref<Form<T, S, F>> | undefined> | undefined'.
                    Type 'Validator<LegacyRef<Form<T, S, F>> | undefined>' is not assignable to type 'Validator<Ref<Form<T, S, F>> | undefined>'.
                      Type 'LegacyRef<Form<T, S, F>> | undefined' is not assignable to type 'Ref<Form<T, S, F>> | undefined'.
      ../../node_modules/@mantine/core/lib/components/Modal/use-modals-stack.d.ts(13,40): error TS1139: Type parameter declaration expected.
      ../../node_modules/@mantine/core/lib/components/Modal/use-modals-stack.d.ts(13,48): error TS1005: ',' expected.
      ../../node_modules/@mantine/core/lib/components/Modal/use-modals-stack.d.ts(13,62): error TS1005: ',' expected.
      ../../node_modules/@mantine/core/lib/components/Modal/use-modals-stack.d.ts(13,70): error TS1005: ')' expected.
      ../../node_modules/@mantine/core/lib/components/Modal/use-modals-stack.d.ts(13,74): error TS1011: An element access expression should take an argument.
      ../../node_modules/@mantine/core/lib/components/Modal/use-modals-stack.d.ts(13,75): error TS1005: ';' expected.
      ../../node_modules/@mantine/core/lib/components/Modal/use-modals-stack.d.ts(13,76): error TS1128: Declaration or statement expected.
      npm error Lifecycle script `build:ts` failed with error:
      npm error code 1
      npm error path /projects/react-jsonschema-form/packages/mantine
      npm error workspace @rjsf/mantine@6.0.0-alpha.0
      npm error location /projects/react-jsonschema-form/packages/mantine
      npm error command failed
      npm error command sh -c tsc -b tsconfig.build.json && tsc-alias -p tsconfig.build.json
      npm error Lifecycle script `build` failed with error:
      npm error code 1
      npm error path /projects/react-jsonschema-form/packages/mantine
      npm error workspace @rjsf/mantine@6.0.0-alpha.0
      npm error location /projects/react-jsonschema-form/packages/mantine
      npm error command failed
      npm error command sh -c npm run build:ts && npm run build:cjs && npm run build:esm && npm run build:umd
`

heath-freenome commented:
@zfarhad Just merged another theme. Can you rebase and update?

zfarhad commented:
> @zfarhad Just merged another theme. Can you rebase and update? Also, I see the same build errors when I locally do `npm update`. I'm guessing some internal dependency update is causing this. Hopefully the update to the `package-lock.json` I made in my last PR helps you. Consider reverting the `package-lock.json` in your MR and just doing an `npm install` again after you merge. Also, I updated the linters so you may have to run `npm run lint -- --fix` in your branch after rebasing

I've rebased and updated the branch. Also fixed the lint errors and warnings.

But I still get errors on build! I think the problem comes from the utils package, when I just try to build it I have these errors:
`

> @rjsf/utils@6.0.0-alpha.0 build
> npm run build:ts && npm run build:cjs && npm run build:esm && npm run build:umd


> @rjsf/utils@6.0.0-alpha.0 build:ts
> npm run compileReplacer && rimraf ./lib && tsc -b tsconfig.build.json && tsc-alias -p tsconfig.build.json


> @rjsf/utils@6.0.0-alpha.0 compileReplacer
> tsc -p tsconfig.replacer.json && move-file lodashReplacer.js lodashReplacer.cjs

src/ErrorSchemaBuilder.ts:43:45 - error TS2590: Expression produces a union type that is too complex to represent.

43     let errorBlock: ErrorSchema = hasPath ? get(this.errorSchema, pathOfError) : this.errorSchema;
                                               ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

src/getOptionMatchingSimpleDiscriminator.ts:31:25 - error TS2339: Property 'type' does not exist on type 'JSONSchema7Definition | {}'.
  Property 'type' does not exist on type 'false'.

31       if (discriminator.type === 'object' || discriminator.type === 'array') {
                           ~~~~

src/getOptionMatchingSimpleDiscriminator.ts:31:60 - error TS2339: Property 'type' does not exist on type 'JSONSchema7Definition | {}'.
  Property 'type' does not exist on type 'false'.

31       if (discriminator.type === 'object' || discriminator.type === 'array') {
                                                              ~~~~

src/getOptionMatchingSimpleDiscriminator.ts:35:25 - error TS2339: Property 'const' does not exist on type 'JSONSchema7Definition | {}'.
  Property 'const' does not exist on type 'false'.

35       if (discriminator.const === value) {
                           ~~~~~

src/getOptionMatchingSimpleDiscriminator.ts:39:25 - error TS2339: Property 'enum' does not exist on type 'JSONSchema7Definition | {}'.
  Property 'enum' does not exist on type 'false'.

39       if (discriminator.enum?.includes(value)) {
                           ~~~~

src/schema/getMatchingOption.ts:50:29 - error TS2345: Argument of type '{} | Exclude<NonNullable<S["properties"]>[string], undefined>' is not assignable to parameter of type 'S'.
  'S' could be instantiated with an arbitrary type which could be unrelated to '{} | Exclude<NonNullable<S["properties"]>[string], undefined>'.

50       if (validator.isValid(discriminator, value, rootSchema)) {
                               ~~~~~~~~~~~~~

src/optionsList.ts:73:15 - error TS2322: Type 'NonNullable<S["properties"]>[string] | undefined' is not assignable to type 'S'.
  'S' could be instantiated with an arbitrary type which could be unrelated to 'NonNullable<S["properties"]>[string] | undefined'.

73         const innerSchema: S = get(aSchema, [PROPERTIES_KEY, selectorField]);
                 ~~~~~~~~~~~

src/schema/getDefaultFormState.ts:444:54 - error TS2322: Type 'NonNullable<S["properties"]>[string]' is not assignable to type 'object'.
  Type 'JSONSchema7Definition' is not assignable to type 'object'.
    Type 'boolean' is not assignable to type 'object'.

444           ((isObject(propertySchema) && CONST_KEY in propertySchema) || hasParentConst) &&
                                                         ~~~~~~~~~~~~~~

src/schema/getDefaultFormState.ts:444:54 - error TS18048: 'propertySchema' is possibly 'undefined'.

444           ((isObject(propertySchema) && CONST_KEY in propertySchema) || hasParentConst) &&
                                                         ~~~~~~~~~~~~~~

src/schema/getDefaultFormState.ts:446:36 - error TS2345: Argument of type 'JSONSchema7Definition | undefined' is not assignable to parameter of type 'JSONSchema7'.
  Type 'undefined' is not assignable to type 'JSONSchema7'.

446           !constIsAjvDataReference(propertySchema);
                                       ~~~~~~~~~~~~~~

src/schema/getDefaultFormState.ts:449:69 - error TS2345: Argument of type 'NonNullable<S["properties"]>[string] | undefined' is not assignable to parameter of type 'S'.
  'S' could be instantiated with an arbitrary type which could be unrelated to 'NonNullable<S["properties"]>[string] | undefined'.

449         const computedDefault = computeDefaults<T, S, F>(validator, propertySchema, {
                                                                        ~~~~~~~~~~~~~~

src/schema/sanitizeDataForNewSchema.ts:96:11 - error TS2322: Type '{} | Exclude<NonNullable<S["properties"]>[string], undefined>' is not assignable to type 'S'.
  'S' could be instantiated with an arbitrary type which could be unrelated to '{} | Exclude<NonNullable<S["properties"]>[string], undefined>'.

96       let oldKeyedSchema: S = get(oldSchema, [PROPERTIES_KEY, key], {});
             ~~~~~~~~~~~~~~

src/schema/sanitizeDataForNewSchema.ts:97:11 - error TS2322: Type '{} | Exclude<NonNullable<(S & { properties: Record<string, unknown> extends S | undefined ? (S | undefined)[never] : unknown; } & { [uniqueSymbol]: unknown; })["properties"]>[string], undefined>' is not assignable to type 'S'.
  'S' could be instantiated with an arbitrary type which could be unrelated to '{} | Exclude<NonNullable<(S & { properties: Record<string, unknown> extends S | undefined ? (S | undefined)[never] : unknown; } & { [uniqueSymbol]: unknown; })["properties"]>[string], undefined>'.

97       let newKeyedSchema: S = get(newSchema, [PROPERTIES_KEY, key], {});
             ~~~~~~~~~~~~~~

src/schema/toIdSchema.ts:81:9 - error TS2345: Argument of type 'NonNullable<S["properties"]>[string] | {} | undefined' is not assignable to parameter of type 'S'.
  'S' could be instantiated with an arbitrary type which could be unrelated to 'NonNullable<S["properties"]>[string] | {} | undefined'.

81         isObject(field) ? field : {},
           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~

src/schema/toPathSchema.ts:149:9 - error TS2345: Argument of type 'NonNullable<S["properties"]>[string] | undefined' is not assignable to parameter of type 'S'.
  'S' could be instantiated with an arbitrary type which could be unrelated to 'NonNullable<S["properties"]>[string] | undefined'.

149         field,
            ~~~~~


Found 15 errors.

`

heath-freenome commented on packages/mantine/package.json:
```diff
@@ -0,0 +1,125 @@
+    "@babel/plugin-proposal-optional-chaining": "^7.21.0",
+    "@babel/preset-env": "^7.23.9",
+    "@babel/preset-react": "^7.23.3",
+    "@babel/preset-typescript": "^7.23.3",
+    "@rjsf/core": "^6.0.0-alpha.0",
```
Why don't you have the three `@mantine` libraries included here as dev dependencies?

heath-freenome replied:
You will need them for the tests right?

heath-freenome commented:
@zfarhad what version of node are you using? What happens if you first checkout `rjsf-v6`, do `npm refresh-node-modules` and then switch to your branch, remove the `package-lock.json` and then do `npm install` again? Does this solve the issue?

heath-freenome commented on packages/mantine/package.json:
```diff
@@ -0,0 +1,125 @@
+    "@rjsf/core": "^6.0.0-alpha.0",
+    "@rjsf/snapshot-tests": "^6.0.0-alpha.0",
+    "@rjsf/utils": "^6.0.0-alpha.0",
+    "@rjsf/validator-ajv8": "^6.0.0-alpha.0",
+    "@types/jest": "^29.5.12",
+    "@types/lodash": "^4.14.202",
+    "@types/react": "^18.2.58",
+    "@types/react-dom": "^18.2.19",
+    "@types/react-test-renderer": "^18.0.7",
+    "atob": "^2.1.2",
+    "babel-jest": "^29.7.0",
+    "eslint": "^8.56.0",
+    "jest": "^29.7.0",
+    "jest-environment-jsdom": "^29.7.0",
+    "nanoid": "^3.3.7",
+    "react": "^18.2.0",
+    "react-dom": "^18.2.0",
+    "react-test-renderer": "^18.2.0",
+    "rimraf": "^5.0.5",
+    "rollup": "^3.29.4",
+    "typescript": "^4.9.5"
```
I just merged a `package.json` cleanup PR that eliminates the need for these:
```suggestion
    "@rjsf/validator-ajv8": "^6.0.0-alpha.0",
    "atob": "^2.1.2",
    "eslint": "^8.56.0",
    "nanoid": "^5.1.5"
```

heath-freenome commented on packages/mantine/package.json:
```diff
@@ -0,0 +1,125 @@
+    "@mantine/hooks": ">=7",
+    "@mantine/dates": ">=7"
+  },
+  "devDependencies": {
+    "@babel/core": "^7.23.9",
+    "@babel/plugin-proposal-class-properties": "^7.18.6",
+    "@babel/plugin-proposal-optional-chaining": "^7.21.0",
+    "@babel/preset-env": "^7.23.9",
+    "@babel/preset-react": "^7.23.3",
+    "@babel/preset-typescript": "^7.23.3",
+    "@rjsf/core": "^6.0.0-alpha.0",
```
```suggestion
  "devDependencies": {
    "@rjsf/core": "^6.0.0-alpha.0",
```

heath-freenome commented:
@zfarhad hopefully when you rebase onto the latest `rjsf-v6` branch (or main if I merge it down before you respond) then hopefully your build issues will go away since I updated all the libraries and fixed the typing issues

heath-freenome commented:
@zfarhad sorry about the close. You'll have to rebase on main now and change the merge to be main

zfarhad commented:
@heath-freenome 
I've rebased and updated the branch source to `rjsf-team:main`
Now all build issues have been solved.

heath-freenome commented on .github/workflows/ci.yml:
```diff
@@ -2,9 +2,9 @@ name: CI
   push:
-    branches: [main]
+    branches: [main, rjsf-v6]
   pull_request:
-    branches: [main]
+    branches: [main, rjsf-v6]
```
You can revert these changes now that the `rjsf-v6` branch is merged and deleted

heath-freenome commented on .github/workflows/codeql.yml:
```diff
@@ -13,9 +13,9 @@ name: "CodeQL"
   push:
-    branches: [ "main" ]
+    branches: [ "main", "rjsf-v6" ]
   pull_request:
-    branches: [ "main" ]
+    branches: [ "main", "rjsf-v6" ]
```
You can revert these changes now that the `rjsf-v6` branch is merged and deleted

heath-freenome commented on CHANGELOG.md:
Can you revert this file and then add a line for `6.0.0-beta.7` with the addition of your new theme with something like:

```
# 6.0.0-beta.6

## @rjsf/mantine

- Added new theme!

```

heath-freenome commented on CHANGELOG_v6.md:
This file was deleted, go ahead and remove it

heath-freenome commented on package.json:
```diff
@@ -101,5 +102,12 @@
     "packages/shadcn"
-  ]
+  ],
+  "dependencies": {
+    "@restart/hooks": "^0.6.2",
+    "@restart/ui": "^1.9.4",
+    "ajv": "^8.17.1",
+    "ajv-keywords": "^5.1.0",
+    "uncontrollable": "^9.0.0"
+  }
```
These belong as `devDependencies` in your theme's `package.json`. If you need the `@restart` stuff as dependencies in your theme, do that

heath-freenome commented on packages/core/test/ArrayField.test.jsx:
```diff
@@ -1745,6 +1745,41 @@ describe('ArrayField', () => {
       );
     });
 
+    it('should handle a change event with multiple files that results the same items in the list', async () => {
```
I'm curious why you added this test?

heath-freenome commented on packages/docs/docs/api-reference/form-props.md:
You can revert these changes as they already exist in the file

heath-freenome commented on packages/docs/docs/usage/themes.md:
```diff
@@ -5,16 +5,17 @@ meaning that you must load the Bootstrap stylesheet on the page to view the form
+| antd                  | Published     | `@rjsf/antd`            |
+| Bootstrap 3 (default) | Published     | `@rjsf/core`            |
+| react-bootstrap       | Published     | `@rjsf/react-bootstrap` |
+| Chakra UI             | Published     | `@rjsf/chakra-ui`       |
+| fluentui-rc           | Published     | `@rjsf/fluentui-rc`     |
```
Oops, we forgot to add the `daisyui` here before:
```suggestion
| Chakra UI             | Published     | `@rjsf/chakra-ui`       |
| Daisy UI              | Published     | `@rjsf/daisyui`         |
| fluentui-rc           | Published     | `@rjsf/fluentui-rc`     |
```

heath-freenome commented on packages/docs/docs/usage/themes.md:
```diff
@@ -5,16 +5,17 @@ meaning that you must load the Bootstrap stylesheet on the page to view the form
+| react-bootstrap       | Published     | `@rjsf/react-bootstrap` |
+| Chakra UI             | Published     | `@rjsf/chakra-ui`       |
+| fluentui-rc           | Published     | `@rjsf/fluentui-rc`     |
+| mantine               | Not Published | `@rjsf/mantine`         |
```
We'll have to remember to update this to Published when we merge the theme

heath-freenome commented on packages/mantine/README.md:
```diff
@@ -0,0 +1,153 @@
+- `@mantine/hooks >= 7`
+- `@mantine/dates >= 7`
+- `dayjs >= 1.8.0`
+- `@rjsf/core >= 2.0.0`
```
Hmmm, either we update all themes that have this, or we just remove it.
```suggestion
- `@rjsf/core >= 6.0.0`
- `@rjsf/utils >= 6.0.0`
- `@rjsf/validator-ajv8 >= 6.0.0`
```

heath-freenome commented on packages/mantine/package.json:
```diff
@@ -0,0 +1,107 @@
+  "description": "Mantine theme, fields and widgets for react-jsonschema-form",
+  "exports": {
+    ".": {
+      "require": "./dist/index.js",
+      "import": "./lib/index.js",
+      "types": "./lib/index.d.ts"
```
A warning happens when `types` isn't first
```suggestion
      "types": "./lib/index.d.ts",
      "require": "./dist/index.js",
      "import": "./lib/index.js"
```

heath-freenome commented on packages/mantine/package.json:
```diff
@@ -0,0 +1,107 @@
+      "types": "./lib/index.d.ts"
+    },
+    "./lib": {
+      "require": "./dist/index.js",
+      "import": "./lib/index.js",
+      "types": "./lib/index.d.ts"
```
```suggestion
      "types": "./lib/index.d.ts",
      "require": "./dist/index.js",
      "import": "./lib/index.js"
```

heath-freenome commented on packages/mantine/package.json:
```diff
@@ -0,0 +1,107 @@
+      "types": "./lib/index.d.ts"
+    },
+    "./lib/*.js": {
+      "require": "./dist/*.js",
+      "import": "./lib/*.js",
+      "types": "./lib/*.d.ts"
```
```suggestion
      "types": "./lib/*.d.ts",
      "require": "./dist/*.js",
      "import": "./lib/*.js"
```

heath-freenome commented on packages/mantine/package.json:
```diff
@@ -0,0 +1,107 @@
+      "types": "./lib/*.d.ts"
+    },
+    "./dist": {
+      "require": "./dist/index.js",
+      "import": "./lib/index.js",
+      "types": "./lib/index.d.ts"
```
```suggestion
      "types": "./lib/index.d.ts",
      "require": "./dist/index.js",
      "import": "./lib/index.js"
```

heath-freenome commented on packages/mantine/package.json:
```diff
@@ -0,0 +1,107 @@
+      "types": "./lib/index.d.ts"
+    },
+    "./dist/*.js": {
+      "require": "./dist/*.js",
+      "import": "./lib/*.js",
+      "types": "./lib/*.d.ts"
```
```suggestion
      "types": "./lib/*.d.ts",
      "require": "./dist/*.js",
      "import": "./lib/*.js"
```

heath-freenome commented on packages/mantine/package.json:
```diff
@@ -0,0 +1,107 @@
+    ]
+  },
+  "peerDependencies": {
+    "@rjsf/core": "^6.x",
+    "@rjsf/utils": "^6.x",
```
```suggestion
    "@rjsf/core": "^6.0.0-beta",
    "@rjsf/utils": "^6..0.0-beta",
```

heath-freenome commented on packages/mantine/package.json:
```diff
@@ -0,0 +1,107 @@
+    "@rjsf/utils": "^6.0.0-beta.1",
+    "@rjsf/validator-ajv8": "^6.0.0-beta.1",
+    "eslint": "^8.56.0",
+    "react-bootstrap": "^2.10.9",
```
Do you need `react-bootstrap`?

heath-freenome commented on packages/mantine/src/templates/ArrayFieldItemTemplate.tsx:
```diff
@@ -0,0 +1,40 @@
+  );
+
+  return (
+    <Box key={`array-item-${index}`} className={className || 'array-item'} mb='xs'>
```
```suggestion
    <Box key={`rjsf-array-item-${index}`} className={className || 'rjsf-array-item'} mb='xs'>
```

heath-freenome commented on packages/mantine/src/templates/ArrayFieldTemplate.tsx:
```diff
@@ -0,0 +1,103 @@
+        <Group justify='flex-end'>
+          <AddButton
+            id={buttonId<T>(idSchema, 'add')}
+            className='array-item-add'
```
We changed the marker classes to add `rjsf-` prefix
```suggestion
            className='rjsf-array-item-add'
```

heath-freenome commented on packages/mantine/src/templates/DescriptionField.tsx:
```diff
@@ -0,0 +1,23 @@
+import { DescriptionFieldProps, FormContextType, RJSFSchema, StrictRJSFSchema } from '@rjsf/utils';
+import { Text } from '@mantine/core';
```
```suggestion
import { DescriptionFieldProps, FormContextType, RJSFSchema, StrictRJSFSchema } from '@rjsf/utils';
import { RichDescription } from '@rjsf/core';
import { Text } from '@mantine/core';
```

heath-freenome commented on packages/mantine/src/templates/DescriptionField.tsx:
```diff
@@ -0,0 +1,23 @@
+  S extends StrictRJSFSchema = RJSFSchema,
+  F extends FormContextType = any,
+>(props: DescriptionFieldProps<T, S, F>) {
+  const { id, description } = props;
+  if (description) {
+    return (
+      <Text id={id} mt={3} mb='sm'>
+        {description}
+      </Text>
+    );
+  }
+
```
Use the new `RichDescription` from `core`
```suggestion
  const { id, description, registry, uiSchema } = props;
  if (description) {
    return (
      <Text id={id} mt={3} mb='sm'>
        <RichDescription description={description} registry={registry} uiSchema={uiSchema} />
      </Text>
    );
  }

```

heath-freenome commented on packages/mantine/src/templates/FieldTemplate.tsx:
```diff
@@ -0,0 +1,71 @@
+      {...otherProps}
+    >
+      {children}
+      {displayLabel && rawDescription && (
+        <Text size='sm' mt={5}>
+          {description}
+        </Text>
+      )}
```
The `SchemaField` renders the `DescriptionTemplate` as `description`, not sure you want it double wrapped in `Text`:

```suggestion
      {displayLabel && rawDescription && description}
```

heath-freenome commented on packages/mantine/src/templates/WrapIfAdditionalTemplate.tsx:
```diff
@@ -0,0 +1,96 @@
+        <RemoveButton
+          id={buttonId<T>(id, 'remove')}
+          iconType='sm'
+          className='array-item-remove'
```
```suggestion
          className='rjsf-array-item-remove'
```

heath-freenome commented on packages/mui/test/__snapshots__/Array.test.tsx.snap:
```diff
@@ -2090,6 +2090,7 @@ label[data-shrink=false]+.MuiInputBase-formControl .emotion-3:focus::-ms-input-p
           onClick={[Function]}
         >
           <div
+            aria-controls=":r3:"
```
Where did these come from?!

heath-freenome commented on packages/playground/src/app.tsx:
```diff
@@ -90,6 +91,47 @@ const themes: PlaygroundProps['themes'] = {
       },
     },
   },
+  'daisy-ui': {
```
Can you delete this since I moved it below and this make it a duplicate

heath-freenome commented on packages/playground/src/app.tsx:
```diff
@@ -151,6 +197,10 @@ const themes: PlaygroundProps['themes'] = {
     stylesheet: '//cdn.jsdelivr.net/npm/bootstrap@5.3.2/dist/css/bootstrap.min.css',
     theme: BootstrapTheme,
   },
+  'react-bootstrap': {
```
This is also duplicated right above so just remove it too

heath-freenome commented:
@zfarhad I hope everything is ok. Let us know when you are ready to get this theme merged

Algirdyz commented:
I would love to have this too!
Seems like a lot of work was put into it already.
Something we could do to finish up?

heath-freenome commented:
@Algirdyz It looks like the author of the PR disappeared. You are welcome to pick up the work in your own PR to get it over the line.

heath-freenome commented:
Closing in favor of https://github.com/rjsf-team/react-jsonschema-form/pull/4692
